acme/localStore.go

```diff
@@ -53,45 +34,73 @@ func (s *LocalStore) Load() (cluster.Object, error) {
 	if err != nil {
 		return nil, err
 	}
-	if err := json.Unmarshal(file, &account); err != nil {
-		return nil, err
+	if len(file) > 0 {
+		if err := json.Unmarshal(file, &account); err != nil {
+			return nil, err
+		}
 	}
-	account.Init()
-	s.account = account
-	log.Infof("Loaded ACME config from store %s", s.file)
 	return account, nil
 }
 
-// Begin creates a transaction with the KV store.
-func (s *LocalStore) Begin() (cluster.Transaction, cluster.Object, error) {
-	s.storageLock.Lock()
-	return &localTransaction{LocalStore: s}, s.account, nil
-}
-
-var _ cluster.Transaction = (*localTransaction)(nil)
-
-type localTransaction struct {
-	*LocalStore
-	dirty bool
-}
+// ConvertToNewFormat converts old acme.json format to the new one and store the result into the file (used for the backward compatibility)
+func ConvertToNewFormat(fileName string) {
+	localStore := acme.NewLocalStore(fileName)
+	storeAccount, err := localStore.GetAccount()
+	if err != nil {
+		log.Warnf("Failed to read new account, ACME data conversion is not unable : %v", err)
+		return
+	}
 
-// Commit allows to set an object in the file storage
-func (t *localTransaction) Commit(object cluster.Object) error {
-	t.LocalStore.account = object.(*Account)
-	defer t.storageLock.Unlock()
-	if t.dirty {
-		return fmt.Errorf("transaction already used, please begin a new one")
+	if storeAccount == nil {
+		localStore := NewLocalStore(fileName)
+		account, err := localStore.Get()
+		if err != nil {
+			log.Warnf("Failed to read old account, ACME data conversion is not unable : %v", err)
```

Comment:
I don't understand the end of your sentence `ACME data conversion is not unable`. 
You maybe mean `ACME data conversion is not available` ?